reframe/core/logging.py

```diff
@@ -132,6 +133,37 @@ def close(self):
             super().close()
 
 
+def _format_time_rfc3339(timestamp, datefmt):
+    tz_suffix = time.strftime('%z', timestamp)
+    tz_rfc3339 = tz_suffix[:-2] + ':' + tz_suffix[-2:]
+
+    # Python < 3.7 truncates the `%`, whereas later versions don't
+    return re.sub(r'(%)?\:z', tz_rfc3339, time.strftime(datefmt, timestamp))
+
+
+class RFC3339Formatter(logging.Formatter):
+    def formatTime(self, record, datefmt=None):
+        datefmt = datefmt or self.default_time_format
+        if '%:z' not in datefmt:
+            return super().formatTime(record, datefmt)
+        else:
+            timestamp = self.converter(record.created)
+            return _format_time_rfc3339(timestamp, datefmt)
+
+    def format(self, record):
+        datefmt = self.datefmt or self.default_time_format
+        try:
+            if record.check_job_completion_time is not None:
+                ct = self.converter(record.check_job_completion_time)
+                record.check_job_completion_time = _format_time_rfc3339(
+                    ct, datefmt
+                )
+        except AttributeError:
```

Comment:
Most probably, I don't need the `try/except` here, because we always define all the extras.